source/common/event/user_space_file_event_impl.h

```diff
@@ -72,9 +69,11 @@ class EventListenerImpl : public EventListener {
 // A FileEvent implementation which is used to drive BufferedIoSocketHandle.
 class UserSpaceFileEventImpl : public FileEvent, Logger::Loggable<Logger::Id::io> {
 public:
+  UserSpaceFileEventImpl(Event::Dispatcher& dispatcher, Event::FileReadyCb cb, uint32_t events);
+
   ~UserSpaceFileEventImpl() override {
-    if (schedulable_.enabled()) {
-      schedulable_.cancel();
+    if (schedulable_->enabled()) {
+      schedulable_->cancel();
```

Comment:
You can rely on destructor to do unregistration of the schedulable_.  You can remove the explicit destructor implementation for UserSpaceFileEventImpl

Reply:
:+1:
Fixing